### PR TITLE
updated options to resize the window on "ok" clic, added keybindings …

### PR DESCRIPTION
**Pull request by sbordeyne**
…to copy elements to clipboard, added help file and menu, download path saved, fixed the window opening just once, games list now properly refreshes on database update through add_game ui

### CODE REVIEW

**Comment by cyian-1756** [comment]
Trying to download a game now errors out with 

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.5/tkinter/__init__.py", line 1553, in __call__
    return self.func(*args)
  File "gui.py", line 200, in download_selected_game
    self.downloaded_games[gamename] = version
NameError: name 'gamename' is not defined
```

**Comment by sbordeyne** [comment]
yes I fixed that
I missed it when I refactored the code

**Comment by sbordeyne** [comment]
It should be all good now

**Comment by cyian-1756** [comment]
There's an issue with downloaded file names on *nix systems 

Trying to download a game without first setting a download loction ends up with the downloaded file being named "GUI\$FILENAME" (When downloading the game Insexsity the gui names the end file "GUI\0.265_Linux_64_(U).zip"

File: GUI/functions.py [hunk 1]
```diff
@@ -19,3 +19,12 @@ def in_database(link):
 		print("No game with that name found")
 	elif link == "Doesn’t support":
 		print("Game doesn't support this OS")
+
+def checkversion(version1, version2):
```
Comment (cyian-1756) [inline GUI/functions.py:24]: This is going to break for a lot of games. Some games use dates as version (jan/10/2017, 7/11/2018) others use month names (December release) and some even use code names. You should add a check to make sure the game uses the standard . system (1.0.0 ect)

**Review by cyian-1756 — CHANGES_REQUESTED** [review]

**Comment by sbordeyne** [comment]
For the first inquiry, it may be because config.cfg is pushed to the repo alongside with the code, and the download path is my download path.

for the other inquiry, I'm going to look into regular expressions, even though I'm bad with them, but that's going to be the cleanest way to check if the version numbering is correct, although, why those devloppers don't use standards is still a mystery to me.

**Comment by sbordeyne** [comment]
these should work, the failsafe is implemented, and that should work, although I didn't extensively test it.

**Comment by sbordeyne** [comment]
I made a release of the current build : https://www.dropbox.com/s/qb8mlm98sqloidr/NSFWGameManager-1.0-windows.zip?dl=1

**Comment by sbordeyne** [comment]
it's windows only though, you need to build it on linux and mac (I used cx_Freeze, that's why)

**Review by cyian-1756 — APPROVED** [review]

**Comment by cyian-1756** [comment]
I'm pretty sure the regex you used (regex = re.compile(r"([\d.]+)")) will match anything starting with a number as . is a wildcard, I'd change it to (regex = re.compile(r"([\d\\.]+)")

> why those devloppers don't use standards is still a mystery to me.

You and me both

> I used cx_Freeze, that's why

I think we should use cx_Freeze/some other py to exe tool for windows and just ship the raw python for Mac and Linux. In my experience the *nix equivalents of python to binary tools tend to produce massive (50MB or larger even for simple program) binaries, also it's much more likely that a linux user will be able to install and run pip

**Comment by sbordeyne** [comment]
> You and me both

I mean, they're called standards for a reason, it's not like version numbering is https://xkcd.com/927/

> I think we should use cx_Freeze/some other py to exe tool for windows and just ship the raw python for Mac and Linux

Yeah, that's not a bad idea, at least for linux users who are a little more tech-savy than windows/mac users (in general). Mac users though, we should make a build for them, but it requires a mac. And I don't have one.

**Comment by cyian-1756** [comment]
> And I don't have one.

Neither do I. An alternative might to be ship an install script that installs pip (if it's not installed) along with all the libs we need

**Comment by sbordeyne** [comment]
you may merge this PR by the way, I'll open another one when I get to work on other features.

For now, this is a stable build, it still needs more testing, but we can't test every scenario ourselves.

Check out the planned features in the updated, main, readme file, and tell me what you think of them.


I think we can start to distribute it as a standalone, maybe get in contact with nsfwgaming and lewdgames mods to feature the tool in the sidebar ?

**Comment by cyian-1756** [comment]
> I think we can start to distribute it as a standalone, maybe get in contact with nsfwgaming and lewdgames mods to feature the tool in the sidebar ?

Yea that's a good idea.

**Comment by sbordeyne** [comment]
I r/lewdgames mods are into it.

**Comment by cyian-1756** [comment]
> I r/lewdgames mods are into it.

You mind rewording that? I can't quite understand it

**Comment by sbordeyne** [comment]
they are willing to link this software in the sidebar.

**Comment by cyian-1756** [comment]
Sweet

**Pull request merged**